### PR TITLE
Fix Issue #46410 Onyx show facts module does not work for onyx versio…

### LINKED ISSUE #46410: Onyx show facts module does not work for onyx version 3.6.6000 and higher
##### SUMMARY
failed to run onyx_show_facts module because of changed in json formmat.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible/lib/ansible/modules/network/onyx/onyx_facts.py

##### ANSIBLE VERSION

```
ansible 2.6.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)] 
```
##### STEPS TO REPRODUCE

```
- hosts: switch132
  gather_facts: no
  connection: network_cli
  become: yes
  vars: 
    ansible_network_os: onyx
  tasks:
    - onyx_facts:
         gather_subset:
         - all
     - onyx_facts:
         gather_subset:
           - "!modules"
     - onyx_facts:
         gather_subset:
           - interfaces
           - modules
```

##### EXPECTED RESULTS
attached yml file should be passed successfully when run it!

##### ACTUAL RESULTS
got unexpected exception

### PR DESCRIPTION
**Pull request by anasbadaha**
…n 3.6.6000 and higher

Signed-off-by: Anas Badaha <anasb@mellanox.com>

##### SUMMARY

Fixes #46410 Onyx show facts module does not work for onyx version 3.6.6000 and higher

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

onyx_facts.py
##### ANSIBLE VERSION

```
ansible 2.6.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)] 

```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/network/onyx/onyx_facts.py [hunk 1]
```diff
@@ -196,25 +196,38 @@ def populate(self):
...
+                for if_name, interface_data in iteritems(if_data):
+                    interface_data = interface_data[0]
+                    if_dict["MAC Address"] = interface_data["Mac address"]
+                    if_dict["Actual Speed"] = interface_data["Actual speed"]
+                    if_dict["MTU"] = interface_data["MTU"]
+                    if_dict["Admin State"] = interface_data["Admin state"]
+                    if_dict["Operational State"] = interface_data["Operational state"]
+                    if_name = if_dict["Interface Name"] = if_name
+            else:
+                if_name = if_dict["Interface Name"] = if_data["header"]
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_facts.py:230]: you have a duplicate code here.
extracting values from json dict should be done once.
Add a function def _extractIfData(interface_data)
that should get all interface data except the interface name

**Review by samerd — CHANGES_REQUESTED** [review]

**Comment by ganeshrn** [comment]
@anasbadaha The PR contains merge commit. Please rebase the PR, refer https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html for steps to rebase

**Comment by anasbadaha** [comment]
Hi @ganeshrn,
I pulled megre commits please merge this PR.

Anas Badaha

**Comment by ganeshrn** [comment]
bot_status

**Comment by ganeshrn** [comment]
The PR still has merge commit and changes requested by @samerd
Can you fix the issues with PR

**Comment by anasbadaha** [comment]
Will push it in new pull request!

**Pull request closed without merging**